front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,436 @@
+import {
+  Avatar,
+  Button,
+  CloudArrowLeftRightIcon,
+  Dialog,
+  Icon,
+  LockIcon,
+  Modal,
+  Page,
+} from "@dust-tt/sparkle";
+import type {
+  ConnectorProvider,
+  ConnectorType,
+  DataSourceIntegration,
+  DataSourceType,
+  EditedByUser,
+  UpdateConnectorRequestBody,
+  UserType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { CONNECTOR_TYPE_TO_MISMATCH_ERROR } from "@dust-tt/types";
+import { InformationCircleIcon } from "@heroicons/react/20/solid";
+import type { NextRouter } from "next/router";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import type { PostManagedDataSourceRequestBody } from "@app/pages/api/w/[wId]/data_sources/managed";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+interface DataSourceEditionModalProps {
+  isOpen: boolean;
+  owner: WorkspaceType;
+  connectorProvider: ConnectorProvider | null | undefined;
+  dataSourceIntegration: DataSourceIntegration | DataSourceType | null;
```

Comment:
I'm not sure to see which call site passes a `DataSourceIntegration` but I'll try to ensure that this component only accept `DataSourceType`.

Reply:
Now that we only display this modal within the `data-sources/[name]` view, we can actually remove it yes 👍🏼

Reply:
Well, this `Modal` is supposed to also handle connection creation, right? so in this case you would get a `DataSourceIntegration`